### ISSUE TITLE
Incompatible with next React version

### ISSUE THREAD

**Issue by kassens** [excalidraw/excalidraw#7895]
The Excalidraw bundles `react/jsx-runtime`, but not `react` itself. This means that if a new React version changes the interface between jsx and react, it'll break.

You can see this by searching for `__SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED` in 
https://unpkg.com/@excalidraw/excalidraw@0.17.5/dist/excalidraw.production.min.js
That internal was renamed in https://github.com/facebook/react/commit/f6131653570bbbf62d642ba9343b9cd0ab1ae97c

What should be done here is to configure the Excalidraw build such that `react/jsx-runtime` is not bundled and remain a `require` just like `react` or `react-dom`.

**Comment by dwelle** [comment]
Hi. The `0.17.x` is a UMD build, and React does not publish a UMD build of `react/jsx-runtime`, which means that consumers that do not use ESM wouldn't be able to use our package.

As such, removing `react/jsx-runtime` from the bundle was postponed until we publish an ESM package, which is currently being released under the unstable `latest` tag (upcoming `0.18.x`).

/cc @ad1992 in case I got the details wrong 🙏

**Comment by ad1992** [comment]
Yes Since there is no umd build available for react-jsxruntime and won't be in the future as well in favor of [ESM](https://github.com/facebook/react/issues/20923) hence making the umd build work isn't possible hence its bundled for `umd`.
 If you try with `next` version (`yarn add @excalidraw/excalidraw@next`), it's ESM and `react-jsxruntime` and it doesn't get bundled.
As @dwelle mentioned above, we will soon ship `0.18.x`

**Comment by nishanthkarthik** [comment]
Thank you for considering a new release - is there a timeline for 0.18.x? I know I can use `next` for now but I'd prefer using the release version in production.

**Comment by ad1992** [comment]
@nishanthkarthik we will be releasing soon in the next couple of weeks.

**Comment by eMerzh** [comment]
@ad1992 don't want to push here, new version of react was released more than a month ago, is there any news on the release of next front?
